src/main/java/teammates/ui/webapi/action/GetFeedbackSessionAction.java

```diff
@@ -28,16 +30,25 @@ public void checkSpecificAccessControl() {
         switch (intent) {
         case STUDENT_SUBMISSION:
         case STUDENT_RESULT:
+            if (userInfo == null && StringHelper.isEmpty(getRequestParamValue(Const.ParamsNames.REGKEY))) {
+                throw new UnauthorizedAccessException("Instructor or Student account is required to access this resource.");
+            }
             StudentAttributes studentAttributes = getStudentOfCourseFromRequest(courseId);
             checkAccessControlForStudentFeedbackSubmission(studentAttributes, feedbackSession);
             break;
         case FULL_DETAIL:
+            if (userInfo == null) {
```

Comment:
Shouldn't we use `gateKeeper.verifyLoggedInUserPrivileges();` for any null check for `userInfo`?